### PR TITLE
fix(no-store-subscription): only handle store subscriptions

### LINKED ISSUE #175: False positive for no-store-subscription
## Behavior

It highlights not only subscriptions to the store, but just any subscription if a store is involved.

## Code snippet

```
    this.formCtrlOrg.valueChanges.pipe(takeUntil(this.drop)).subscribe((orgName) => {
      this.store.dispatch(UserPageActions.checkOrgNameInput({orgName}));
    });
```

### PR DESCRIPTION
**Pull request by rafaelss95**
Fixes #175.

### CODE REVIEW

File: src/rules/store/no-store-subscription.ts [hunk 1]
```diff
@@ -31,9 +31,7 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
     if (!storeName) return {}
 
     return {
-      [`CallExpression:has(${storeExpression(
-        storeName,
-      )}) > MemberExpression > Identifier[name='subscribe']`](
+      [`CallExpression:matches([callee.object.callee.object.name=${storeName}], [callee.object.callee.object.object.type='ThisExpression'][callee.object.callee.object.property.name=${storeName}]) > MemberExpression > Identifier[name='subscribe']`](
```
Comment (rafaelss95) [inline src/rules/store/no-store-subscription.ts:34]: Unfortunately I don't think we can reuse the `storeExpression` function anymore, but let me know if you have any suggestions.

File: tests/rules/no-store-subscription.test.ts [hunk 2]
```diff
@@ -11,17 +11,45 @@ ruleTester().run(path.parse(__filename).name, rule, {
       @Component()
```
Comment (rafaelss95) [inline tests/rules/no-store-subscription.test.ts:11]: Do you know how could I test a single file with `uvu`? Also, is there any way to check what test file is not behaving as expected? Currently I got this:

Captura de Tela 2021-08-30 às 10 57 54

Comment (timdeschryver) [reply tests/rules/no-store-subscription.test.ts:11]: Yikes, a single test can be run with the following command.
Sadly it doesn't support a glob.

```
node -r ts-node/register ./tests/rules/no-store-subscription.test.ts
```

For the AssertionError, I think we'll have to wrap the testrunner to also print the file 🤔
I'll try to work on this soon.

File: src/rules/store/no-store-subscription.ts (hunk 1 above)
Comment (timdeschryver) [reply src/rules/store/no-store-subscription.ts:34]: I think you're right.
Does this mean that we can remove `storeExpression` ?

**Review by timdeschryver — APPROVED** [review]
LGTM, let's get this merged! 🚀

**Pull request merged**